src/components/Form.js

```diff
@@ -39,14 +40,18 @@ export default class Form extends Component {
     const mustValidate = edit && !props.noValidate && liveValidate;
     const { definitions } = schema;
     const formData = getDefaultFormState(schema, props.formData, definitions);
+
+    //create a retrievedSchema object here to get new idSchema
```

Comment:
I don't think this comment is necessary.

Reply:
Okay, made the necessary changes. Let me know if anything else needs to be done.